src/backend/api/Fusion.Resources.Api/Controllers/Departments/DepartmentsController.cs

```diff
@@ -79,6 +79,27 @@ public async Task<ActionResult> AddDelegatedResourceOwner(string departmentStrin
             return CreatedAtAction(nameof(GetDepartments), new { departmentString }, null);
         }
 
+        [HttpDelete("/departments/{departmentString}/delegated-resource-owner/{azureUniqueId}")]
+        public async Task<IActionResult> DeleteDelegatedResourceOwner(string departmentString, Guid azureUniqueId)
+        {
+            #region Authorization
+            var authResult = await Request.RequireAuthorizationAsync(r =>
+            {
+                r.AlwaysAccessWhen().FullControl().FullControlInternal();
+            });
+
+            if (authResult.Unauthorized)
+                return authResult.CreateForbiddenResponse();
+            #endregion
+
+            var deleted = await DispatchAsync(
+                new DeleteDelegatedResourceOwner(departmentString, azureUniqueId)
+            );
+
+            if (deleted) return Ok();
+            else return NotFound();
```

Comment:
Any possibility to check for existing data and return NotFound before issuing DeleteDelegatedResourceOwner ?

I would have checked for data before auth check returning NotFound if nothing to remove.

Reply:
I don't think disclosing whether an item exists or not to unauthorized users is necessary, in most cases it does not matter, but in some cases even knowing something is there could be sensitive info.